src/code.cloudfoundry.org/vxlan-policy-agent/converger/converger.go

```diff
@@ -165,42 +166,67 @@ func (m *SinglePollCycle) SyncASGsForContainer(containers ...string) error {
 	}
 
 	var cleanupDuration time.Duration
-
 	if len(containers) == 0 {
 		cleanupStart := time.Now()
-
-		deletedChains, err := m.enforcer.CleanChainsMatching(regexp.MustCompile(planner.ASGManagedChainsRegex), desiredChains)
+		err := m.cleanupASGsChains(planner.ASGManagedChainsRegex, desiredChains)
 		if err != nil {
-			errors = multierror.Append(errors, fmt.Errorf("clean-up-orphaned-asg-chains: %s", err))
-		} else {
-			m.logger.Debug("policy-cycle-asg", lager.Data{
-				"message": "deleted-orphaned-chains",
-				"chains":  deletedChains,
-			})
-
-			for chainKey, chainName := range m.containerToASGChain {
-				for _, deletedChain := range deletedChains {
-					if deletedChain.Table == chainKey.Table && deletedChain.Name == chainName {
-						delete(m.containerToASGChain, chainKey)
-						delete(m.asgRuleSets, chainKey)
-					}
-				}
-			}
+			errors = multierror.Append(errors, err)
 		}
 		cleanupDuration = time.Now().Sub(cleanupStart)
 	}
 	m.asgMutex.Unlock()
 
-	pollDuration := time.Now().Sub(pollStartTime)
 	m.metricsSender.SendDuration(metricASGEnforceDuration, enforceDuration)
+
 	if cleanupDuration != 0 {
 		m.metricsSender.SendDuration(metricASGCleanupDuration, cleanupDuration)
 	}
+	pollDuration := time.Now().Sub(pollStartTime)
 	m.metricsSender.SendDuration(metricASGPollDuration, pollDuration)
 
 	return errors
 }
 
+func (m *SinglePollCycle) CleanupOrphanedASGsChains(containerHandle string) error {
+	m.asgMutex.Lock()
+	defer m.asgMutex.Unlock()
+
+	var cleanupDuration time.Duration
+	cleanupStart := time.Now()
+	var err error
+	err = m.cleanupASGsChains(planner.ASGChainPrefix(containerHandle), []enforcer.LiveChain{})
+	cleanupDuration = time.Now().Sub(cleanupStart)
+
+	if cleanupDuration != 0 {
+		m.metricsSender.SendDuration(metricASGOrphanCleanupDuration, cleanupDuration)
```

Comment:
Do we want to emit this metric when cleaning up a single ASG chain?

Reply:
We no longer do this as of abe7f26